### PR TITLE
PCI-1724 Add integration tests

### PR DESCRIPTION
**Pull request by aliculPix4D**
This PR contains only:
- (first commit) refactoring of dependabot main script without any behavior changes
- (first commit) add integration tests for the main script
- (second commit) add regression test (failing because main code tries to fetch wrong branch)
- (third commit) pass the regression test

### CODE REVIEW

**Comment by Pix4D-Janus** [comment]
Your pipeline is set.
Check it here:https://builder.ci.pix4d.com/teams/developers/pipelines/dependabot-PCI-1724-integration-tests

File: docker/dependabot-main.rb [hunk 1]
```diff
@@ -42,89 +34,123 @@ def requirements(checker)
...
             new(source: source, credentials: credentials_github)
 
   files = fetcher.files
   commit = fetcher.commit
+
+  [files, commit]
+end
+
+def file_parser(files, source)
   # Parse the dependency files
```
Comment (odormond) [inline docker/dependabot-main.rb:49]: So it's not a parser but the parsed result? If so then the function's name is a bit confusing.

File: docker/dependabot-main.rb [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-# DOCKER_PASS
-
 require "dependabot/docker"
 require "dependabot/auto_merge"
 require "dependabot/pr_info"
 require "dependabot/path_level"
 
 def pr_creator(source, commit, updated_deps, updated_files, credentials_github)
-  Dependabot::PullRequestCreator.new(
+  # Create a pull request for the update
```
Comment (odormond) [inline docker/dependabot-main.rb:9]: Does it return the created PR? If so it's not a creator but a PR: `pr_creator` -> `create_pr`.

File: docker/dependabot-main.rb [hunk 3]
```diff
@@ -42,89 +34,123 @@ def requirements(checker)
...
-  )
+def file_fetcher(source, credentials_github)
   # Fetch the dependency files
   fetcher = Dependabot::FileFetchers.for_package_manager("docker").
             new(source: source, credentials: credentials_github)
 
   files = fetcher.files
   commit = fetcher.commit
+
+  [files, commit]
```
Comment (odormond) [inline docker/dependabot-main.rb:45]: Again, this looks like the fetched results and not a fetcher: `file_fetcher` -> `fetch_files_and_commit`.

File: docker/dependabot-main.rb [hunk 4]
```diff
@@ -42,89 +34,123 @@ def requirements(checker)
...
-      dependency: dep, dependency_files: files, credentials: credentials_docker
-    )
-    next if checker.up_to_date?
+def file_updater(dep, updated_deps, files)
+  # Generate updated dependency files
+  print "  - Updating #{dep.name} (from #{dep.version}) \n"
+  updater = Dependabot::FileUpdaters.for_package_manager("docker").new(
+    dependencies: updated_deps, dependency_files: files, credentials: nil
+  )
+  updater.updated_dependency_files
```
Comment (odormond) [inline docker/dependabot-main.rb:62]: ... `file_updater` -> `update_files`

**Review by odormond — COMMENTED** [review]
Function names should tell the reader either what they do or what they return but not what they use. This is an implementation detail and the purpose of the function is to hide it.

File: docker/dependabot-main.rb (hunk 2 above)
Comment (aliculPix4D) [reply docker/dependabot-main.rb:9]: Yes, it creates a PR and returns the PR data as a dictionary/hash

File: docker/dependabot-main.rb (hunk 3 above)
Comment (aliculPix4D) [reply docker/dependabot-main.rb:45]: Done as suggested.

File: docker/dependabot-main.rb (hunk 4 above)
Comment (aliculPix4D) [reply docker/dependabot-main.rb:62]: Done as suggested.

File: docker/dependabot-main.rb (hunk 1 above)
Comment (aliculPix4D) [reply docker/dependabot-main.rb:49]: It actually parse the files and return a list of dependencies. I have change it to `fetch_dependencies`.

**Review by odormond — APPROVED** [review]

**Comment by marco-m-pix4d** [comment]
The author of the commits is Pix4D-Janus; could you please `git commit --amend` and use yourself instead?

**Review by marco-m-pix4d — APPROVED** [review]

**Comment by aliculPix4D** [comment]
> The author of the commits is Pix4D-Janus; could you please `git commit --amend` and use yourself instead?

Sure. I don't know how that happened. I will correct it now. Thanks for pointing this to me.

**Pull request merged**